InnerEye/ML/SSL/lightning_containers/ssl_container.py

```diff
@@ -0,0 +1,243 @@
+                             SSLDatasetName.CheXpert.value: CheXpert}
+
+    ssl_augmentation_config = param.ClassSelector(class_=Path, allow_None=True,
+                                                  doc="The path to the yaml config defining the parameters of the "
+                                                      "augmentations. Ignored for CIFAR10 example")
+    ssl_training_dataset_name = param.ClassSelector(class_=SSLDatasetName, doc="The name of the dataset")
+    ssl_training_batch_size = param.Integer(
+        doc="Total training batch size, will be divided across the number of gpus used for training. For example: if "
+            "you specify ssl_training_batch_size=1600 and use 4 nodes with 4 gpus each (i.e. total of 16 GPUs), "
+            "the code will provide a per-gpu batch size of 100")
+    ssl_training_type = param.ClassSelector(class_=SSLTrainingType, doc="Which algorithm to use for SSL training")
+    ssl_encoder = param.ClassSelector(class_=EncoderName, doc="Which encoder to use for SSL")
+    use_balanced_binary_loss_for_linear_head = param.Boolean(default=False,
+                                                             doc="Whether to use a balanced loss for the training of "
+                                                                 "the linear head")
+    num_workers = param.Integer(default=6, doc="Number of workers to use for dataloader processes.")
+    is_debug_model = param.Boolean(default=False,
+                                   doc="If True, the training will be restricted to 1 batch per epoch."
+                                       "Used for debugging and tests.")
+    classifier_augmentation_config = param.ClassSelector(class_=Path,
+                                                         doc="The path to the yaml config for the linear head "
+                                                             "augmentations")
+    classifier_dataset_name = param.ClassSelector(class_=SSLDatasetName,
+                                                  doc="Name of the dataset to use for the linear head training")
+    classifier_batch_size = param.Integer(default=256, doc="Batch size for linear head tuning")
+    online_evaluator_lr = param.Number(default=1e-4, doc="Learning rate for linear head training during SSL training.")
+
+    def setup(self) -> None:
+        from InnerEye.ML.SSL.lightning_containers.ssl_image_classifier import SSLClassifierContainer
+        self.total_num_gpus = self.num_gpus_per_node * self.num_nodes
+        self._load_config()
+        # If you're using the same data for training and linear head, allow the user to specify the dataset only
+        # once. Or if you are doing just finetuning of linear head, the user should be able to specify dataset via
+        # azure_dataset_id/local_dataset instead of extra_dataset fields (as in this case we only use one dataset).
+        if ((self.classifier_dataset_name == self.ssl_training_dataset_name) or isinstance(self,
```

Comment:
I got lost in here. What is the purpose of this condition and why do we enter this code path for finetuning?

Reply:
Discussed it offline. Fine-tuning container inherits from this class. We will add this info in the docstring - they both share the same `setup` method.